outwatch/src/main/scala/outwatch/dom/Handler.scala

```diff
@@ -1,34 +1,28 @@
 package outwatch
+package dom
 
-import monix.execution.{Ack, Scheduler}
-import monix.reactive.subjects.PublishSubject
+import cats.effect.Sync
+import cats.implicits._
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.observers.Subscriber
+import monix.reactive.subjects.{BehaviorSubject, ReplaySubject}
 import monix.reactive.{Observable, Observer}
 
 import scala.concurrent.Future
 
-trait MonixOps {
-  type ProHandler[-I, +O] = Observable[O] with Observer[I]
-  type Handler[T] = ProHandler[T,T]
+trait HandlerOps[F[_]] {
+  object Handler {
+    def empty[T](implicit F: Sync[F]): F[Handler[T]] = create[T]
 
-  implicit class RichObserver[I](observer: Observer[I]) {
-    def redirect[I2](f: Observable[I2] => Observable[I]): ConnectableObserver[I2] = {
-      val subject = PublishSubject[I2]
-      val transformed = f(subject)
-      new ConnectableObserver[I2](subject, implicit scheduler => transformed.subscribe(observer))
-    }
-
-    @deprecated("Use contramap", "")
-    def redirectMap[I2](f: I2 => I): Observer[I2] = observer.contramap(f)
+    def create[T](implicit F: Sync[F]): F[Handler[T]] = F.delay(unsafe[T])
+    def create[T](seed:T)(implicit F: Sync[F]): F[Handler[T]] = F.delay(unsafe[T](seed))
 
-    def redirectMapMaybe[I2](f: I2 => Option[I]): Observer[I2] = new Observer[I2] {
-      override def onNext(elem: I2): Future[Ack] = f(elem).fold[Future[Ack]](Ack.Continue)(observer.onNext(_))
-      override def onError(ex: Throwable): Unit = observer.onError(ex)
-      override def onComplete(): Unit = observer.onComplete()
-    }
-
-    def redirectCollect[I2](f: PartialFunction[I2, I]): Observer[I2] = redirectMapMaybe(f.lift)
-    def redirectFilter(f: I => Boolean): Observer[I] = redirectMapMaybe(e => Some(e).filter(f))
+    def unsafe[T]: Handler[T] = ReplaySubject.createLimited(1)
+    def unsafe[T](seed:T): Handler[T] = BehaviorSubject[T](seed)
   }
+}
+
+trait ProHandlerOps[F[_]] extends HandlerOps[F] {
 
   implicit class RichProHandler[I,O](self: ProHandler[I,O]) {
```

Comment:
Fixed in 1cf9400. The `ProHandler.create` method did require a context, but `ProHandler.apply` and ProHandler.connectible did not. It appears to me that `.apply` and `.connectable` are internal/private api methods, so I put those methods in a separate private object. Let me know if you think these are methods users would need, but I think all users would create via `ProHandler.create`.

Reply:
Nice! I think, you can make the `ProHandlerBuilder`, because user might want to extend on our custom monix ops, where there might need to construct a `ProHandler`. 

How about naming this object `ProHandler` again and then add extension methods to `ProHandler` in the `ProHandlerOps`? We could do the same for `Handler`, so have the basic `unsafe` method at least.
```
implicit class ProHandlerMethods(factory: ProHandler.type) { def create[T] = ... }
```

Reply:
At this point, given the other comments, we've deleted ProHandlerOps, so I think we can just revert the ProHandlerBuilder

Reply:
Reverted it in 02d41b0.